Feat: Update Discover-More (#114)
NRCHKB/NRCHKB.github.io: content/wiki/discover-more/experimental/index.md

toc: true
---

> These are beta versions containing possible future features.
> We use beta versions to test and improve our features without compromising the experience of our users.

Author: ```suggestion
This mode provides access to possible future features for experienced users. These features are not yet fully tested or documented, may not be the final solution, and are subject to change at any time...
```

> {{< alert icon="‼️" text="Beta versions may contain bug fixes and improvements!" />}}

Reviewer: It is not special version :)
It is shipped in production release available via experimental mode.

> {{< alert icon="‼️" text="Beta versions may contain bug fixes and improvements!" />}}
> 

Author: ```suggestion
{{< alert icon="‼️" text="Using experimental mode is not recommended for new users!" />}}

```

## Open Node-RED in beta mode

In order to be able to use experimental features you have to start Node-RED with `NRCHKB_EXPERIMENTAL=true` environment variable.

Enter this command in your terminal:

```bash
sudo NRCHKB_EXPERIMENTAL=true node-red
```

With `DEBUG` in addition:

```bash
sudo NRCHKB_EXPERIMENTAL=true DEBUG=NRCHKB* node-red
```

> {{< alert icon="💡" text="Don't forget to stop Node-RED first before launching it in beta mode." />}}

Reviewer: Same
```suggestion
{{< alert icon="💡" text="Don't forget to stop Node-RED first before launching it in experimental mode." />}}
```

## Use of Debug

Using debug allows you to see logs and possible errors.

Below is a table with all the possible logs and their description:

| Options | Description |
|---|---|
| NRCHKB* | Show all HK logs |
| ServiceUtils | Show Service node logs |
| BridgeUtils | Show Bridge logs |
| AccessoryUtils | Show Accessory logs |
| CharacteristicUtils | Show Characteristic logs |
| MdnsUtils | Show Mdns logs |

## Start automatically in beta mode

> If you want to start Node-RED in beta mode when your system restarts:

Reviewer: ```suggestion
If you want to start Node-RED in experimental mode when your system restarts:
```

> contributors: ["Shaquu","crxporter"]

Author: ```suggestion
contributors: ["Shaquu","crxporter","GogoVega"]
```

```

### Service 2

Added `Get` functionality so that the node can retrieve attribute values when a refresh is requested. [issues#392](https://github.com/NRCHKB/node-red-contrib-homekit-bridged/issues/392)

Together with NRCHKB 2.0.0 (TBC) `onSet` and `onChange` will be merged into unified output called events.

It will output data for all event types defined below:

- set
- change
- subscribe
- unsubscribe
- characteristic-warning

Node output will look like:

> ```js

Reviewer: ```suggestion
```ts
```
